### PR TITLE
BrefKernel return types

### LINKED ISSUE #46: Not type-hinting return values of various KernelInterface methods is deprecated in Symfony 5.4
I'm receiving the following deprecations using Symfony 5.4-BETA.

>  1x: Method "Symfony\Component\HttpKernel\KernelInterface::getCacheDir()" might add "string" as a native return type declaration in the future. Do the same in implementation "Bref\SymfonyBridge\BrefKernel" now to avoid errors or add an explicit @return annotation to suppress this message.

>  1x: Method "Symfony\Component\HttpKernel\KernelInterface::getLogDir()" might add "string" as a native return type declaration in the future. Do the same in implementation "Bref\SymfonyBridge\BrefKernel" now to avoid errors or add an explicit @return annotation to suppress this message.

>  1x: Method "Symfony\Component\HttpKernel\HttpKernelInterface::handle()" might add "Response" as a native return type declaration in the future. Do the same in implementation "Bref\SymfonyBridge\BrefKernel" now to avoid errors or add an explicit @return annotation to suppress this message.

Update the return types in the following `Bref\SymfonyBridge\BrefKernel` methods:
- `getCacheDir(): string`
- `getLogDir(): string`
- `handle():: Response`

### PR DESCRIPTION
**Pull request by amacrobert**
Fixes #46

### CODE REVIEW

**Comment by t-richard** [comment]
Note that this is a breaking change.

**Comment by amacrobert** [comment]
Should I address the PHPCS failure, even though it's from code not in my PR?

**Comment by chekalsky** [comment]
@mnapoli any plans on merging this? AFAIK those deprecations are critical in order to use Symfony 6?

**Comment by mnapoli** [comment]
How can we solve the breaking change?

Is it a matter of upgrading the Symfony dependency? Is so, what would be the new range?

**Comment by t-richard** [comment]
Actually I think this just needs a new minor tag because of the BC break. 

Defining a return type on a child method where a parent does not define one is allowed (covariance of return types) which means we can still support older versions of Symfony.

So it would require to
[x] Add types where needed
[ ] Allow `symfony/http-kernel:^6.0` and `symfony/filesystem:^6.0` (fixes #49)
[ ] Update the CI to run tests on newer PHP & Symfony versions
[ ] Merge into master and tag version `0.2.0`

This way we remove deprecation notices for Symfony 5.4, are compatible with Symfony 6.0 and ensure PHP 8.1 compatibility.

We may want to split into multiple PRs but I think it would make sense to do all of this before tagging `0.2.0`.

Does that make sense?

**Comment by mnapoli** [comment]
Sounds good to me

**Comment by amacrobert** [comment]
Alright I updated composer dependencies, and added latest PHP and Symfony versions to the tests. Awaiting workflow approval.

**Review by t-richard — APPROVED** [review]
Seems good, let's just wait for the tests to run.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,8 +10,8 @@ jobs:
       max-parallel: 10
       fail-fast: false
       matrix:
-        php: ['7.3', '7.4', '8.0']
-        sf_version: ['4.4.*', '5.0.*', '5.2.*']
+        php: ['7.3', '7.4', '8.0', '8.1']
+        sf_version: ['4.4.*', '5.0.*', '5.2.*', '5.4.*', '6.0.*']
```
Comment (t-richard) [inline .github/workflows/ci.yml:14]: Symfony 6 is compatible with php >= 8.0 so running tests on 7.3 and 7.4 will always fail

```suggestion
        sf_version: ['4.4.*', '5.0.*', '5.2.*', '5.4.*', '6.0.*']
        exclude:
          - php: 7.3
            sf_version: 6.0.*
            
          - php: 7.4
            sf_version: 6.0.*
```

**Review by t-richard — CHANGES_REQUESTED** [review]
Tests are failing on 6.0 because Bref is a dependency of this bridge and is not yet compatible because of Silly (see https://github.com/brefphp/bref/pull/1117)

**Comment by mnapoli** [comment]
Thanks, merged via #50!

**Pull request merged**